tests/ctst/steps/quotas/quotas.ts

```
@@ -0,0 +1,150 @@
+                    await createBucketWithConfiguration(world, scenarioWithExampleID,
+                        isBucketNonVersioned ? '' : 'with');
+                    await putObject(world);
+                    output[scenarioWithExampleID] = {
+                        AccessKey: CacheHelper.parameters?.AccessKey || Constants.DEFAULT_ACCESS_KEY,
+                        SecretKey: CacheHelper.parameters?.SecretKey || Constants.DEFAULT_SECRET_KEY,
+                    };
+                }
+            }
+        }
+    
+        await createJobAndWaitForCompletion(world, 'end2end-ops-count-items', 'quotas-setup');
+        await Utils.sleep(2000);
+        fs.writeFileSync(filePath, JSON.stringify({
+            ready: true,
+            ...output,
+        }));
+    
+        await releaseLock();
+    } else {
+        while (!fs.existsSync(filePath)) {
+            await Utils.sleep(100);
+        }
+
+        let configuration: { ready: boolean } = JSON.parse(fs.readFileSync(filePath, 'utf8')) as { ready: boolean };
+        while (!configuration.ready) {
+            await Utils.sleep(100);
+            configuration = JSON.parse(fs.readFileSync(filePath, 'utf8')) as { ready: boolean };
+        }
+    }
+
+    const configuration: typeof output = JSON.parse(fs.readFileSync(`/tmp/${featureName}`, 'utf8')) as typeof output;
+    const key = hashStringAndKeepFirst20Characters(`${pickle.astNodeIds[1]}`);
+    world.parameters.logger?.debug('Scenario key', { key, from: `${pickle.astNodeIds[1]}`, configuration });
+    const config = configuration[key];
+    world.resetGlobalType();
+    // Save the bucket name for the scenario
+    world.addToSaved('bucketName', key);
+    // Save the account name for the scenario
+    Zenko.saveAccountAccessKeys(config.AccessKey, config.SecretKey);
+    world.addToSaved('accountName', key);
+});
+
+Given('a bucket quota set to {int} B', async function (this: Zenko, quota: number) {
+    if (quota === 0) {
```

```suggestion
    if (quota <= 0) {
```

---

Or can you set "uint" in the sentence definition ?

---

I don't want to add this condition, as <= should throw an error from the API: this is something we might want to test someday. 0 is for no quota, which is exactly what this step shouldn't do

There is also no uint support

---

>  this is something we might want to test someday. 

I feel like this is the same for 0 quotas in that case, we might want to test that, setting 0 quotas actually removes the quotas limit, what do you think ?

---

Oh, no, because we have dedicated functions for that, as we have separate permissions: deleteAccountQuota (or bucket), 0 is not a valid value for the update API. We could decide to call the delete dynamically if needed, but for now it's not used...

---

OK 👌